addons/sale/static/src/js/tours/product_configurator_tour_utils.js

```diff
@@ -0,0 +1,315 @@
+    `;
+}
+
+function optionalProductImageSrc(productName) {
+    return queryAttribute(
+        `${optionalProductSelector(productName)} td.o_sale_product_configurator_img>img`, 'src'
+    );
+}
+
+function addOptionalProduct(productName) {
+    return {
+        content: `Add ${productName}`,
+        trigger: `
+            ${optionalProductSelector(productName)}
+            td.o_sale_product_configurator_price
+            button:contains("Add")
+        `,
+        run: 'click',
+    };
+}
+
+function increaseProductQuantity(productName) {
+    return {
+        content: `Increase the quantity of ${productName}`,
+        trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_qty
+            button:has(i.fa-plus)
+        `,
+        run: 'click',
+    };
+}
+
+function decreaseProductQuantity(productName) {
+    return {
+        content: `Decrease the quantity of ${productName}`,
+        trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_qty
+            button:has(i.fa-minus)
+        `,
+        run: 'click',
+    };
+}
+
+function setProductQuantity(productName, quantity) {
+    return {
+        content: `Set the quantity of ${productName} to ${quantity}`,
+        trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_qty
+            input[name="product_quantity"]
+        `,
+        run: `edit ${quantity} && click .modal-body`,
+    };
+}
+
+function assertProductQuantity(productName, quantity) {
+    const quantitySelector = `
+        ${productSelector(productName)}
+        td.o_sale_product_configurator_qty
+        input[name="product_quantity"]
+    `;
+    return {
+        content: `Assert that the quantity of ${productName} is ${quantity}`,
+        trigger: quantitySelector,
+        run: async () =>
+            await waitUntil(() => queryValue(quantitySelector) === quantity, { timeout: 1000 }),
+    };
+}
+
+function assertProductOutOfStock(productName) {
+    return {
+        content: `Assert that ${productName} is out of stock`,
+        trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_qty:contains("Out of stock")
+        `,
+        extra_trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_qty:not(:has(input[name="product_quantity"]))
+        `,
+        isCheck: true,
+    };
+}
+
+function assertOptionalProductOutOfStock(productName) {
+    return {
+        content: `Assert that ${productName} is out of stock`,
+        trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_price:contains("Out of stock")
+        `,
+        extra_trigger: `
+            ${productSelector(productName)}
+            td.o_sale_product_configurator_price:not(:has(button:contains("Add")))
+        `,
+        isCheck: true,
+    };
+}
+
```

Comment:
Shouldn't we move the test utils to the module they are used? I mean, it feels weird to have utils to check `strike-through price` or `out of stock` in a module that doesn't implements those features.

Reply:
Hmm, I agree with the idea, but I wonder whether we really want to split those helpers in multiple modules (especially since they're only for tests and don't add any dependencies)? I'm fine with either, so let me know if you have a preference.

Reply:
We might consider stock in the backend configurator in the future, but even if we do, we'd still allow to add the product to the cart, so I doubt those checks make sense outside of `website_sale` :thinking:

Reply:
I moved the `website_sale` and `website_sale_stock` specific helpers to their corresponding modules :)